sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala

```diff
@@ -32,16 +32,24 @@ import org.apache.spark.util.{SerializableConfiguration, Utils}
 object ExternalSource {
 
   lazy val hasSparkAvroJar = {
-    val loader = Utils.getContextOrSparkClassLoader
-
     /** spark-avro is an optional package for Spark, so the RAPIDS Accelerator
      * must run successfully without it. */
-    Try(loader.loadClass("org.apache.spark.sql.v2.avro.AvroScan")) match {
+    Try(ShimLoader.loadClass("org.apache.spark.sql.v2.avro.AvroScan")) match {
```

Comment:
My preference is not to change this logic until we have a grasp of the root cause.

Reply:
Let us check if #5723 changes anything

Reply:
I was thinking the same thing. I can quickly test that